fe/fe-core/src/main/java/com/starrocks/planner/IcebergScanNode.java

```diff
@@ -254,20 +258,40 @@ public void setupScanRangeLocations(DescriptorTable descTbl) throws UserExceptio
             hdfsScanRange.setModification_time(0);
             hdfsScanRange.setFile_format(IcebergApiConverter.getHdfsFileFormat(file.format()).toThrift());
 
-            hdfsScanRange.setDelete_files(task.deletes().stream().map(source -> {
-                TIcebergDeleteFile target = new TIcebergDeleteFile();
-                target.setFull_path(source.path().toString());
-                target.setFile_content(
-                        source.content() == FileContent.EQUALITY_DELETES ? TIcebergFileContent.EQUALITY_DELETES :
-                                TIcebergFileContent.POSITION_DELETES);
-                target.setLength(source.fileSizeInBytes());
-
-                if (source.content() == FileContent.EQUALITY_DELETES) {
-                    throw new StarRocksConnectorException("iceberg equality is not supported");
+            List<TIcebergDeleteFile> deleteFiles = new ArrayList<>();
+            for (DeleteFile deleteFile : task.deletes()) {
+                FileContent content = deleteFile.content();
+                if (content == FileContent.EQUALITY_DELETES) {
+                    List<Integer> taskEqualityFieldIds = deleteFile.equalityFieldIds();
+                    if (taskEqualityFieldIds.isEmpty()) {
+                        continue;
+                    }
+                    if (!currentEqualityIds.isEmpty() && !currentEqualityIds.equals(taskEqualityFieldIds)) {
```

Comment:
why don't we support it

Reply:
We will optimize it in the next work. it will reuse the current codes.